docs/dapp/sapphire/precompiles.md

```diff
@@ -238,3 +255,70 @@ bytes memory digest = abi.encodePacked(keccak256("signed message"));
 bytes memory signature = Sapphire.sign(alg, sk, digest, "");
 require( Sapphire.verify(alg, pk, digest, "", signature) );
 ```
+
+## SHA-512
+
+### Overview
+
+ * Precompile address: `0x0100000000000000000000000000000000000101`
+ * Parameters: `bytes input_data`
+
+Hash the input data with SHA-512, according to [NIST.FIPS.180-4]
+
+:::note SHA-512 is vulnerable to length-extension attacks
```

Comment:
To clarify, this is not vulnerable?

Reply:
SHA-512 is vulnerable, the function below (SHA-512/256) is not.